### PR TITLE
Looker parsing improvements for liquid templating and view/model aliasing

### LINKED ISSUE #17911: Looker does not support liquid templates or aliased views/models
**Affected module**
Ingestion Framework

**Describe the bug**
My team uses templating to adjust our sql table name during development, while leaving the production table as the base condition. 

**To Reproduce**
Ingest LookML with liquid templates in the sql_table_name and observe lineage generation failures

Screenshots or steps to reproduce

**Expected behavior**
I would like the ingestion to try to render the templates with no variables to render the base condition. Likewise, I would like the lineage creation to use the `from_` attribute of a Looker view when it exists or the `view_name` of a model when that exists, regardless of whether there are other joins present in the model (as the current logic requires)

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by sam-mccarty-mavenclinic**
I would like to propose two improvements to resolve issues with Looker ingestion, one with liquid template variables in sql_table_name and another with aliased views and explores

### Describe your changes:

Fixes #17911 

* I added python-liquid as a requirement 
* I added a call to try to render the sql table name as a liquid template before passing it to build the lineage request so that it may render the template as the base condition
* I added a check for a non-empty `from_` attribute for views to replace the `name` while building the lineage to correctly map to the actual view being joined
* I replaced the existing conditions with a check for a non-empty `view_name` on the model so that the underlying view it is based on is included as part of the lineage, regardless of whether there are other views joined

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by sushi30** [comment]
Hi @sam-mccarty-mavenclinic, thank you for the PR.

1. You didn't submit the change to `setup.py` therefore the errors in the CI:

```python
 E   ModuleNotFoundError: No module named 'liquid'
```

2. Can you give an example of this scenario? Maybe an example liquid template that we can test?

**Comment by sushi30** [comment]
Take a look [here](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/tests/unit/topology/dashboard/test_looker.py).

File: ingestion/src/metadata/utils/fqn.py [hunk 1]
```diff
@@ -554,7 +554,7 @@ def split_table_name(table_name: str) -> Dict[str, Optional[str]]:
     """
     # Revisit: Check the antlr grammer for issue when string has double quotes
     # Issue Link: https://github.com/open-metadata/OpenMetadata/issues/8874
-    details: List[str] = split(table_name.replace('"', ""))
+    details: List[str] = split(table_name.replace('"', "").replace("`", ""))
```
Comment (sushi30) [inline ingestion/src/metadata/utils/fqn.py:557]: Can you share why you added this character replacement? The fqn usually handles special characters without problems.

Comment (sam-mccarty-mavenclinic) [reply ingestion/src/metadata/utils/fqn.py:557]: I couldn't figure out whether the backticks that we use to wrap our BigQuery tables were handled in the FQN fucntion when I was initially debugging the ingestion. I will confirm when I am working on setting up the test file whether this is being handled by the fqn

Comment (sam-mccarty-mavenclinic) [reply ingestion/src/metadata/utils/fqn.py:557]: From what I could test, fqn.split() doesn't strip the backticks, however ES might be able to find the table regardless of their inclusion when the fqn is built. I will re-run the ingestion without this to confirm whether it was a red herring

Comment (sam-mccarty-mavenclinic) [reply ingestion/src/metadata/utils/fqn.py]: Seems fine without it, I've dropped the `replace`

**Comment by sushi30** [comment]
@sam-mccarty-mavenclinic I merged the branch with main bc we had some failing tests that were fixed. If the CI is green we can merge this.

**Review by sushi30 — APPROVED** [review]

**Pull request merged**